node/service/src/lib.rs

```diff
@@ -111,6 +114,35 @@ type PartialComponentsResult<Client, Backend> = Result<
 	ServiceError,
 >;
 
+const RELAY_CHAIN_SLOT_DURATION_MILLIS: u64 = 6_000;
+
+thread_local!(static TIMESTAMP: std::cell::RefCell<u64> = const { std::cell::RefCell::new(0) });
```

Comment:
thread_local is not appropriate here as it create a global constant per thread. Please use AtomicU64 instead: https://doc.rust-lang.org/stable/std/sync/atomic/struct.AtomicU64.html